pkg/v1/resource/chartconfig/update.go

```diff
@@ -2,13 +2,53 @@ package chartconfig
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
+	clusterGuestConfig, err := r.toClusterGuestConfigFunc(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	chartConfigsToUpdate, err := toChartConfigs(updateChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if len(chartConfigsToUpdate) > 0 {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "updating chartconfigs")
+
+		clusterConfig, err := prepareClusterConfig(r.baseClusterConfig, clusterGuestConfig)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		g8sClient, err := r.guest.NewG8sClient(ctx, clusterConfig.ClusterID, clusterConfig.Domain.API)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, chartConfigToUpdate := range chartConfigsToUpdate {
+			_, err := g8sClient.CoreV1alpha1().ChartConfigs(metav1.NamespaceSystem).Update(chartConfigToUpdate)
+			if apierrors.IsAlreadyExists(err) {
```

Comment:
On updates this check should not be necessary. We can only update because it exists.

Reply:
Yes, this check isn't needed. I've removed it.